internal/tools/list_test.go

```diff
@@ -0,0 +1,460 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/cache"
+)
+
+func newCallToolRequest(args map[string]any) mcp.CallToolRequest {
+	return mcp.CallToolRequest{
+		Params: struct {
+			Name      string         `json:"name"`
+			Arguments map[string]any `json:"arguments,omitempty"`
+			Meta      *struct {
+				ProgressToken mcp.ProgressToken `json:"progressToken,omitempty"`
+			} `json:"_meta,omitempty"`
+		}{
+			Arguments: args,
+		},
+	}
+}
+
+func TestListStepActions(t *testing.T) {
+	tests := []struct {
+		name    string
+		request mcp.CallToolRequest
+		input   []v1beta1.StepAction
+		output  []v1beta1.StepAction
+	}{
+		{
+			name:    "No namespace, no filters",
+			request: newCallToolRequest(map[string]any{}),
+			input: []v1beta1.StepAction{
+				{ObjectMeta: metav1.ObjectMeta{Name: "sa1"}},
+				{ObjectMeta: metav1.ObjectMeta{Name: "sa2"}},
+			},
+			output: []v1beta1.StepAction{
+				{ObjectMeta: metav1.ObjectMeta{Name: "sa1"}},
+				{ObjectMeta: metav1.ObjectMeta{Name: "sa2"}},
+			},
+		},
+		{
+			name:    "With namespace",
+			request: newCallToolRequest(map[string]any{"namespace": "ns1"}),
+			input: []v1beta1.StepAction{
+				{ObjectMeta: metav1.ObjectMeta{Name: "sa1", Namespace: "ns1"}},
+				{ObjectMeta: metav1.ObjectMeta{Name: "sa2", Namespace: "ns2"}},
+			},
+			output: []v1beta1.StepAction{
+				{ObjectMeta: metav1.ObjectMeta{Name: "sa1", Namespace: "ns1"}},
+			},
+		},
+		{
+			name:    "With label selector",
+			request: newCallToolRequest(map[string]any{"label-selector": "env=prod"}),
+			input: []v1beta1.StepAction{
+				{ObjectMeta: metav1.ObjectMeta{Name: "sa1", Labels: map[string]string{"env": "prod"}}},
+				{ObjectMeta: metav1.ObjectMeta{Name: "sa2", Labels: map[string]string{"env": "dev"}}},
+			},
+			output: []v1beta1.StepAction{
+				{ObjectMeta: metav1.ObjectMeta{Name: "sa1", Labels: map[string]string{"env": "prod"}}},
+			},
+		},
+		{
+			name:    "With prefix filter",
+			request: newCallToolRequest(map[string]any{"prefix": "foo"}),
+			input: []v1beta1.StepAction{
+				{ObjectMeta: metav1.ObjectMeta{Name: "foo-sa1"}},
+				{ObjectMeta: metav1.ObjectMeta{Name: "bar-sa2"}},
+			},
+			output: []v1beta1.StepAction{
+				{ObjectMeta: metav1.ObjectMeta{Name: "foo-sa1"}},
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			fakeClient := fakepipelineclientset.NewSimpleClientset()
+			informer := informers.NewSharedInformerFactory(fakeClient, 0)
+			stepActionInformer := informer.Tekton().V1beta1().StepActions()
+			ctx := context.WithValue(t.Context(), stepactioninformer.Key{}, stepActionInformer)
+
+			for _, sa := range test.input {
+				_, err := fakeClient.TektonV1beta1().StepActions(sa.Namespace).Create(t.Context(), &sa, metav1.CreateOptions{})
+				require.NoError(t, err)
+				err = informer.Tekton().V1beta1().StepActions().Informer().GetIndexer().Add(&sa)
+				require.NoError(t, err)
```

Comment:
:thinking: I'm not confident this is the correct way to set things up, but it is working!

Reply:
@vdemeester :eyes: is this the correct way to set up the fake informers?